keccak/src/avx2.rs

```diff
@@ -301,27 +301,24 @@ fn round(i: usize, state: [__m256i; 25]) -> [__m256i; 25] {
     flatten(state)
 }
 
-fn keccak_perm(buf: &mut [__m256i; 25]) {
-    let mut state = *buf;
+fn keccak_perm(buf: &mut [[u64; VECTOR_LEN]; 25]) {
+    let mut state: [__m256i; 25] = unsafe { transmute_copy(buf) };
     for i in 0..24 {
         state = round(i, state);
     }
-    *buf = state;
+    *buf = unsafe { transmute_copy(&state) };
```

Comment:
FWIW, [nomicon](https://doc.rust-lang.org/nomicon/transmutes.html) really doesn't like `transmute_copy`. but I'm not sure how much I trust it these days, and it seems fine to me...

Reply:
fair! i really should just have done a `transmute(state)` here